client/blocks/inline-help/popover.jsx

```diff
@@ -333,6 +333,7 @@ function mapStateToProps( state ) {
 	const postType = getEditedPostValue( state, siteId, postId, 'type' );
 	const gutenbergUrl = getGutenbergEditorUrl( state, siteId, postId, postType );
 	const isEligibleForChecklist = isEligibleForDotcomChecklist( state, siteId );
+	const showSwitchEditorButton = currentRoute.match( /^\/(block-editor|post|page)\// );
```

Comment:
Non-blocking but we can define the regex once, if we move it to a const above the class function. ( A regex object will also have better browser support, though we have polyfills for String methods in Calypso). eg
```jsx
const EDITOR_PATH_REGEX = /^\/(block-editor|post|page)\//;

class InlineHelpPopover extends Component { 
// ...
//.... in map state to props
const showSwitchEditorButton = EDITOR_PATH_REGEX.test( currentRoute );
```